torch/_functorch/partitioners.py

```diff
@@ -452,7 +426,11 @@ def reordering_to_mimic_autograd_engine(gm):
     # Populate depth for the nodes. Depth is the distance from the inputs.
     depths = {}
     output_node = next(node for node in gm.graph.nodes if node.op == "output")
-    get_depth(output_node, depths)
+    for node in gm.graph.nodes:
+        if node.op == 'placeholder':
+            depths[node] = 0
+        else:
+            depths[node] = max([depths[arg] for arg in node.all_input_nodes if isinstance(arg, torch.fx.Node)], default=0)
```

Comment:
`for arg in node.all_input_nodes if isinstance(arg, torch.fx.Node)`

dont think you need the isinstance here with `node.all_input_nodes`